### ISSUE TITLE
Current location deleted when editing asset

### ISSUE THREAD

**Issue by docmedicus-support** [grokability/snipe-it#13889]
### Debug mode

- [X] I have enabled debug mode
- [X] I have read [checked the Common Issues page](https://snipe-it.readme.io/docs/common-issues)

### Describe the bug

As described in https://github.com/snipe/snipe-it/discussions/13682 we almost always loose the current location when editing assets. We set the current location using asset audits to track the assets actual location. When making changes to the asset, the current location seems to get lost and the asset reverts to the default location (if set) or no location.

Is this intended behaviour? We currently have to audit all assets again after editing them just to update the current location.

It would also be great if there was another way to set the current location.

### Reproduction steps

1. Audit asset and set location
2. Edit asset and Save

### Expected behavior

The current location is kept through asset edits.

### Snipe-IT Version

6.1.2

### Operating System

Ubuntu

### Web Server

Apache

### PHP Version

8.1.2

**Comment by Godmartinz** [comment]
Hi @docmedicus,

Are you on the latest version? 

You will want to make sure that the checkbox below is checked if you want the asset's current location to change permanently. If you don't mark this it will just make a note of the location in the audit log. 

<img width="600" alt="image" src="https://github.com/snipe/snipe-it/assets/47435081/5b199ff5-211d-4c5c-b80e-18052e3aaf91">

Let me know if you have any more questions or concerns.

**Comment by docmedicus-support** [comment]
Hi @Godmartinz ,

yes, I have now recreated the problem on the latest version. We do set the location via auditing and checking "Update Asset Location", since this seems to be the only way to set the current location. (why is that?)

If I then edit the asset (e.g. adding supplier informaton and purchase date), the current location is removed from the asset (I can also see this in the assets history).

I assume that either the current location is removed, or reset to the default location (which we do not set any more, because we see no practical use for it). Either way I do not understand the behaviour. Is this intended? We are currently silently loosing asset locations when not manually checking them after each asset edit.

**Comment by docmedicus-support** [comment]
After having a quick look at the code I suspect the following line(s) to be the cause of my confusion:

https://github.com/snipe/snipe-it/blob/776b16d37936a7c36f67fae6404478c28fa095d7/app/Http/Controllers/Assets/AssetsController.php#L152

If the asset is update and not currently checked out, the current location is reset to the default location. What is the reasoning behind this? Can an asset not be at a certain location (as determined by an audit) and then be edited (e.g. to add an invoice). Why does the current location have to be changed in this case?

Also, why is it not possible to set the current location apart from forcing it through an audit?

**Comment by NMX-Drew** [comment]
I'd like to +1 this issue too. having to re-set the location after every edit or sign in/out is not ideal.

**Comment by DrekiDegga** [comment]
I think the idea is that if an item is not checked out to a user, it would be "on the shelf" at it's default location, no?

If you store the item somewhere when it isn't in use, shouldn't that be it's default location?

**Comment by snipe** [comment]
Yeah, I think this is working as intended. If it’s not checked out, it should fall back to its default location.
